drivers/wifi/eswifi/eswifi.h

```diff
@@ -71,6 +72,8 @@ struct eswifi_dev {
 	u8_t mac[6];
 	char buf[MAX_DATA_SIZE];
 	struct k_mutex mutex;
+	atomic_val_t mutex_owner;
+	unsigned int mutex_depth;
```

Comment:
Can't you use semaphore instead here as that does counting automatically?

Reply:
well, yes, but I there is no limit to the depth, really only need this for counting, maybe a semaphore is overkill here,